src/Tickets/Commerce/Admin/Singular_Order_Notices.php

```diff
@@ -0,0 +1,240 @@
+				'The message after an order status was changed to refunded and refunded by the payment gateway.',
+				'event-tickets'
+			),
+			/* translators: %1$s: order number. %2$s: the currency code, e.g. USD. %3$s: the amount refunded. */
+			self::ORDER_PARTIALLY_REFUNDED            => _x(
+				'Order #%1$s has been partially refunded %2$s %3$s',
+				'The message after an order status was changed to refunded and partially refunded by the payment gateway.',
+				'event-tickets'
+			),
+			/**
+			 * Error messages.
+			 */
+			/* translators: %1$s: order number. %2$s: the current status. %3$s: the status to apply. */
+			self::ORDER_STATUS_CHANGE_FAILED          => _x(
+				'Order #%1$s could not be changed from %2$s to %3$s. Please try again or get help.',
+				'The message after an order status failed to change.',
+				'event-tickets'
+			),
+			/* translators: %1$s: order number. */
+			self::ORDER_REFUND_FAILED                 => _x(
+				'Order #%1$s could not be marked as refunded. Please try again. You can still refund directly in the payment gateway.',
+				'The message after an order refund failed.',
+				'event-tickets'
+			),
+			/* translators: %1$s: order number. %2$s: payment gateway. */
+			self::ORDER_GATEWAY_REFUND_FAILED         => _x(
+				'Order #%1$s could not be refunded via %2$s. Please try again or contact us for help.',
+				'The message after an order refund failed in the gateway.',
+				'event-tickets'
+			),
+			/* translators: %1$s: refund amount. %2$s: order number. %3$s: payment gateway. */
+			self::ORDER_GATEWAY_PARTIAL_REFUND_FAILED => _x(
+				'The partial refund of %1$s could not be processed for Order #%2$s via %3$s.',
+				'The message after a partial order refund failed in the gateway.',
+				'event-tickets'
+			),
+		];
+
+		return self::$generated_messages;
+	}
+
+	/**
+	 * Registers the admin notice to be displayed.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $message_code The message key to be displayed.
+	 * @param mixed  ...$params    [optional] Dynamic params for the message.
+	 */
+	public function do_message( string $message_code, ...$params ) {
+		$success = in_array( $message_code, $this->success_messages );
+		$slug    = 'tec-tickets-commerce-order-status-update-notice';
+
+		Tribe__Admin__Notices::instance()->undismiss( $slug );
+
+		tribe_transient_notice(
+			$slug,
+			$this->get_message( $message_code, ...$params ),
+			[
+				'type'     => $success ? 'success' : 'error',
+				'dismiss'  => true,
+				'action'   => 'admin_notices',
+				'priority' => 1,
+			],
+			1
+		);
+	}
+
+	/**
+	 * Get the parsed message string.
+	 *
+	 * @since TBD
+	 *
+	 * @param string $message_code The key for the admin notice.
+	 * @param mixed  ...$params Dynamic params for the message.
+	 *
+	 * @return string
+	 */
+	public function get_message( string $message_code, ...$params ) {
+		$message = sprintf( $this->messages()[ $message_code ], ...$params );
```

Comment:
```suggestion
		$message = sprintf( $this->messages()[ $message_code ], ...array_map( 'esc_html', $params ) );
```

Reply:
We don't want [double escape](https://github.com/the-events-calendar/event-tickets/pull/3210#discussion_r1738757205).